java-sdk/README.asciidoc

```diff
@@ -0,0 +1,212 @@
+
+=== Project Layout
+This documentation and examples use Gradle for building connectors. Other build tools can be used instead.
+Check out the https://docs.gradle.org/current/userguide/tutorial_java_projects.html[Gradle quickstart^] for Java Projects.
+
+The minimal project layout is as follows:
+
+    my-connector
+    ├── build.gradle
+    ├── gradle.properties
+    ├── settings.gradle
+    └── src
+        └── main
+            └── java
+                └── Plugin.java
+                └── Config.java
+                └── Fetcher.java
+
+==== build.gradle
+This is a typical Gradle build file. A few important settings are required. You can have a look
+at an example https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/build.gradle[here^].
+
+==== gradle.properties
+The build utility used for packaging up a connector depends on this file and contains required metadata about the connector,
+including its name, version, and the version of Fusion it can connect to.
+Have a look at the example https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/gradle.properties[here^].
+
+==== src/main/java
+This is where Java code lives by default. This location can be changed, but in this documentation we use the Gradle default.
+
+=== The Plugin
+The `Plugin` class describes the various subcomponents of a connector, as well as its dependencies.
+The current set of subcomponents include the `Fetcher`, validation, security filtering, and configuration suggestions.
+
+You can think of the `Plugin` class as the component that glues everything together for an implementation.
+
+You can see an example
+https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/src/main/java/com/lucidworks/fusion/connector/plugin/RandomContentPlugin.java[here^].
+Notice that each subcomponent can have a set of unique dependencies, or several subcomponents can share dependencies.
+
+=== The Configuration
+The configuration component is an interface, which extends a base `Model` interface. By simply adding `getter` methods
+and `@Property` annotations, you can dynamically generate a type-safe configuration object from a connector's configuration data.
+Similarly, you can generate a Fusion-compatible JSON schema by implementing `Model`.
+
+You can see an example of a configuration implementation
+https://github.com/lucidworks/connectors-sdk-resources/blob/master/java-sdk/connectors/random-connector/src/main/java/com/lucidworks/fusion/connector/plugin/RandomContentConfig.java[here^].
+
+More detailed information on the configuration and schema capabilities can be found 
+ifdef::env-github[]
+link:configuration.asciidoc[here].
+endif::[]
+ifndef::env-github[]
+link:/fusion-server/{version}/search-development/getting-data-in/connectors/sdk/java-sdk/java-connector-dev.html#configuration[here].
+endif::[]
+
+=== The Fetcher
```

Comment:
This will need to be updated to reflect changes in how Fetchers are integrated in 4.1.

You can now implement multiple fetchers, each one being equivalent to a "Phase". The "lifecycle" is the same, but preFetch/postFetch are not called for each Phase, they are only called once; before and after fetch:

```
foreach fetcher do
  start()
    preFetch()
      fetch()
    postFetch()
  stop()
end
```